bootloader/src/pyi_global.c

```diff
@@ -127,4 +127,8 @@ pyi_global_printf(const char *fmt, ...)
     vsyslog(LOG_NOTICE, fmt, v);
     va_end(v);
 #endif
+
+    /* Flush the output buffer because we care more about debug output than speed */
+    fflush(stdout);
+    fflush(stderr);
```

Comment:
I think it should be the other way round: We are writing to stderr above, thus I think this should be flushed first. Or flush stdout prior to writing to stderr.